11-cometbls/misbehaviour_handle.go

```diff
@@ -50,30 +50,18 @@ func (ClientState) CheckForMisbehaviour(ctx sdk.Context, cdc codec.BinaryCodec,
 			return true
 		}
 	case *Misbehaviour:
+		// NOTE: `Header_1.Height >= Header_2.Height` is enforced by `ValidateBasic`
+
 		// if heights are equal check that this is valid misbehaviour of a fork
 		// otherwise if heights are unequal check that this is valid misbehavior of BFT time violation
-		if msg.Header_1.GetHeight() == msg.Header_2.GetHeight() {
-			// TODO(aeryz): we need blockid to be in header
-
-			// blockID1, err := tmtypes.BlockIDFromProto(&msg.Header_1.SignedHeader.Commit.BlockID)
-			// if err != nil {
-			// 	return false
-			// }
-
-			// blockID2, err := tmtypes.BlockIDFromProto(&msg.Header_2.SignedHeader.Commit.BlockID)
-			// if err != nil {
-			// 	return false
-			// }
-
-			// // Ensure that Commit Hashes are different
-			// if !bytes.Equal(blockID1.Hash, blockID2.Hash) {
-			// 	return true
-			// }
-			return false
-
-		} else if !msg.Header_1.Time.After(msg.Header_2.Time) {
-			// Header_1 is at greater height than Header_2, therefore Header_1 time must be less than or equal to
-			// Header_2 time in order to be valid misbehaviour (violation of monotonic time).
+		if msg.Header_1.GetHeight().EQ(msg.Header_2.GetHeight()) {
```

Comment:
we don't call `verifyMisbehaviour` because `checkForMisbehaviour` is called after `verifyMisbehaviour` called during `VerifyClientMessage`